testing/trino-product-tests/src/main/java/io/trino/tests/product/hive/TestParquetArrayOfStructsColumn.java

```diff
@@ -0,0 +1,118 @@
+    public void cleanup()
+    {
+        hdfsClient.delete("/user/hive/warehouse/TestParquetArrayOfStructsColumn");
+    }
+
+    @Test(groups = {PARQUET, STORAGE_FORMATS})
+    public void testSelectArrayOfStructsColumn()
+    {
+        /*
+        message custom_schema {
+          optional binary id (STRING);
+          optional group entries (LIST) {
+            repeated group list {
+              optional group element {
+                optional binary name (STRING);
+                optional group before {
+                  optional binary value (STRING); }
+                optional group after {
+                  optional binary value (STRING); }}}}}
+         */
+        onTrino().executeQuery("CREATE TABLE events (\n" +
+                "entries ARRAY(ROW(name VARCHAR, before ROW(value VARCHAR), after ROW(value VARCHAR))))\n" +
+                "WITH (\n" +
+                "external_location = 'hdfs:///user/hive/warehouse/TestParquetArrayOfStructsRow/events',\n" +
+                "format = 'PARQUET');");
+
+        // TODO Match array content, not just length (requires support from tempto).
+        /*
+        Row expected = row(
+                "{name=test-entry-0, before={value=before-0}, after={value=after-0}}",
+                "{name=test-entry-1, before={value=before-1}, after={value=after-1}}",
+                "{name=test-entry-2, before={value=before-2}, after={value=after-2}}");
+         */
+        assertThat(onTrino().executeQuery("SELECT CARDINALITY(entries) as size FROM events"))
+                .containsExactlyInOrder(row(3));
+
+        onTrino().executeQuery("DROP TABLE events");
+    }
+
+    @Test(groups = {PARQUET, STORAGE_FORMATS})
+    public void testSelectArrayOfStructsColumn_ProtoRepeatedGroup()
+    {
+        // The parquet in events-proto-repeated-group.parquet was written from protobuf using schema below.
+        // Note the 'repeated group' - it does not get parsed successfully as parquet format requires a
+        // wrapper of type LIST and a repeated group inside, as in `testSelectArrayOfStructsColumn()`.
+        /*
+        message Event {
+          optional binary id (STRING) = 1;
+          repeated group entries = 2 {
+            optional binary name (STRING) = 3;
+            optional group before = 4 {
+              optional binary value (STRING) = 5; }
+            optional group after = 6 {
+              optional binary value (STRING) = 7; }}}
+         */
+        onTrino().executeQuery("CREATE TABLE events_proto (\n" +
+                "entries ARRAY(ROW(name VARCHAR, before ROW(value VARCHAR), after ROW(value VARCHAR))))\n" +
+                "WITH (\n" +
+                "external_location = 'hdfs:///user/hive/warehouse/TestParquetArrayOfStructsRow/events-proto',\n" +
+                "format = 'PARQUET');");
+
+        assertThat(onTrino().executeQuery("SELECT entries FROM events_proto"))
+                .containsExactlyInOrder(row((Object) null));
```

Comment:
@skrzypo987 the problem this PR highlights is here - "proto-parquet" files are not supported (null is returned instead of content of a column - an array of nested structs). The test above checks that it works correctly with "proper parquet".